execution_engine/src/storage/global_state/lmdb.rs

```diff
@@ -371,4 +425,65 @@ mod tests {
                 .unwrap()
         );
     }
+
+    #[test]
+    fn copy_one_state_to_another() {
```

Comment:
@henrytill this ended up being trickier to move to `operations::testing` than I thought, so I've left it here.  If it's a show stopper I'll go ahead and move it over there...